local/store.go

```diff
@@ -250,32 +236,46 @@ func (s *Store) addImageToTar(tw *tar.Writer, image v1.Image, withName string) e
 	return addTextToTar(tw, manifestJSON, "manifest.json")
 }
 
-func (s *Store) addLayerToTar(tw *tar.Writer, layer v1.Layer) (string, error) {
-	layerDiffID, err := layer.DiffID()
+func (s *Store) addLayerToTar(tw *tar.Writer, layer v1.Layer, blankIdx *int) (string, error) {
+	// If the layer is a previous image layer that hasn't been downloaded yet,
+	// cause ALL the previous image layers to be downloaded by grabbing the ReadCloser.
+	layerReader, err := layer.Uncompressed()
```

Comment:
Previously, we forced the layer to be downloaded by asking for its size.
    But, this has performance implications when you only need the size to construct a manifest.
    Now, we force the layer to be downloaded by opening a ReadCloser to the layer,
    and size only reports if the layer is present on disk, it doesn't cause the layer to be downloaded.